fabtests/multinode/src/core.c

```diff
@@ -53,20 +53,53 @@
 #include <arpa/inet.h>
 #include <assert.h>
 
+char *tx_barrier;
+char *rx_barrier;
+struct fid_mr *mr_barrier;
+struct fi_context2 *barrier_tx_ctx, *barrier_rx_ctx;
+
 struct pattern_ops *pattern;
 struct multinode_xfer_state state;
+struct multi_xfer_method method;
+struct multi_xfer_method multi_xfer_methods[] = {
+	{
+		.name = "send/recv",
+		.send = multi_msg_send,
+		.recv = multi_msg_recv,
+		.wait = multi_msg_wait,
+	},
+	{
+		.name = "rma",
+		.send = multi_rma_write,
+		.recv = multi_rma_recv,
+		.wait = multi_rma_wait,
+	}
+};
 
-static int multinode_setup_fabric(int argc, char **argv)
+static int multi_setup_fabric(int argc, char **argv)
 {
 	char my_name[FT_MAX_CTRL_MSG];
 	size_t len;
-	int ret;
+	int i, ret;
+	struct fi_rma_iov *remote = malloc(sizeof(struct fi_rma_iov));
```

Comment:
1. use sizeof(*variable), instead of sizeof(structure name).
2. Must check for allocation failures
3. Why is this a malloc anyway?